_lectures/lect09.md

```diff
@@ -6,6 +6,305 @@ ready: true
 pdfurl:
 ---
 
+
+
+# HW05 and HW06
+- Both due Sat 2/8 at 11AM
+- https://ucsb-cs8.github.io/w20/work_list/Syllabus has a link to the book in the library reserve
+- Read the book before the lab
+- Practice the concepts: don’t just try to solve the specific problem
+
+# Survey
+- Thank you for the survey!
+- I wish the professor would check email more often
+    - Professor is more active on piazza
+- I am not able to make any open lab hours or ninja skills sessions
+    - Adding more time slots to office hours
+- I wish we could get more into code structure 
+    - Walk through a specific example
+- Survey is not closed yet!
+- Anonymous link on syllabus to submit feedback
+
+# What we will be reviewing today:
++ Creating Functions
++ Using Strings
++ Getting User Input
++ `while` loops
++ `pytest`
++ `random` module
+
+# `pytest` function skeleton:
+```{r}
+def test_function_XXX():
+    assert functino(X) == return_value
+    
+def test_function_YYY():
+	assert_function(Y) == return_value
+```
+- Here we are defining a new function
+- We will always start with `test_`
+- Usually testing our function with multiple inputs
+- Have to differentiate our `pytest` function from one version to another
+    - This is why we define two different functions (XXX and YYY)
+- Floating point values do not behave as ecpected when we check equality (`==`)
+    - This is why we use the `approx()` function
+        - Sets a certain degree of precision
+        - If the answer on the left matches the answer on the right within that degree of precision python will say these two values are equal
+        - This is the **only** time we use the `approx()` function ie. wouldn’t make sense for a string
+
+# "Helper" functions
+- As soon as you are copy and pasting your code and changing only one value, this should be your sign that you need a function!
+- The one value you are changing when you copy and paste is the **input parameter** in your function
+- Advantages:
+    + Your code will be shorter
+    + When debugging, you only need to fix your **one** error rather than scrolling and fixing your error everywhere you pasted it
+    
+# `random` module
+```{r}
+print (“Random”, random.randint(25, 50))
+```
+- 25 is the starting point, 50 is the ending point
+    - This is an inclusive range
+- From the lab: `seed` is an initial value that gets plugged into the formula and predicts every number that comes after it
+
+
+```python
+import random
+print("Random", random.randint(1, 10))
+```
+
+    ('Random', 7)
```

Comment:
Can you please re-format the output to include what is being displayed? I think seeing parentheses and comma-separated parameters might throw them off.

Reply:
I'm not sure I understand what you mean. How would you like the output formatted?

Reply:
Since parentheses and commas are not being output and there is no `print` before them, it might be confusing as to what that line represents. The output of it (if there was a `print` beforehand) looks different in IDLE (as you discovered, it's not the same in Jupyter notebooks).